hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityEdgeFetcher.java

```diff
@@ -85,23 +90,37 @@
           EdgeSetGroupRequest edgeSetGroupRequest,
           Map<InteractionResponse, Entity> neighborMap) {
     return interactions
-        .flatMapSingle(
+        .flatMapMaybe(
             response ->
                 this.buildEdge(
-                    edgeSetGroupRequest, response.getInteraction(), neighborMap.get(response)))
+                    edgeSetGroupRequest, response.getInteraction(), key, neighborMap.get(response)))
         .collect(Collectors.toUnmodifiableList())
         .map(ConvertedEdgeResultSet::new)
         .map(edgeResultSet -> Map.entry(key, edgeResultSet));
   }
 
-  private Single<Edge> buildEdge(
-      EdgeSetGroupRequest edgeSetGroupRequest, EntityInteraction response, Entity neighbor) {
+  private Maybe<Edge> buildEdge(
+      EdgeSetGroupRequest edgeSetGroupRequest,
+      EntityInteraction response,
+      org.hypertrace.gateway.service.v1.entity.Entity source,
+      Entity neighbor) {
+    if (isNull(neighbor)) {
+      // This should not happen, but we're being defensive due to some underlying issues with entity
+      // and interaction query mismatches
+      LOG.error(
+          "Expected neighbor for interaction {} from source entity {} missing from response",
```

Comment:
Late for the party. I am not sure I understand the message (not a native tho) but maybe "Expected neighbor for interaction {} from source entity {}. Missing from response" is more easier to read?

Reply:
How about `Response missing neighbor entity for interaction {} from source entity {}`

Reply:
https://github.com/hypertrace/hypertrace-graphql/pull/27